From f59f2b6a82261c79e87d7b9f6ecba8538b8cac3a Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Mon, 30 May 2016 12:18:40 +1200
Subject: [PATCH] Makefile.am: Compile vala unstable tests with
 SECRET_WITH_UNSTABLE=1.

Without this the generated C code compiles without the correct header (warnings
are disabled so this is not obvious). This was causing the test to crash on
Ubuntu (exact cause not diagnosed).

https://bugzilla.gnome.org/show_bug.cgi?id=767002
---
 libsecret/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libsecret/Makefile.am b/libsecret/Makefile.am
index 0601460..7d96774 100644
--- a/libsecret/Makefile.am
+++ b/libsecret/Makefile.am
@@ -282,7 +282,7 @@ SUFFIXES = .vala .vala.o
 test-vala-lang.vala.o: libsecret/test-vala-lang.vala $(VALA_TESTS_VAPIS)
 	$(VALA_V) $(VALAC) --compile $(VALA_TESTS_FLAGS) $<
 test-vala-unstable.vala.o: libsecret/test-vala-unstable.vala $(VALA_TESTS_VAPIS)
-	$(VALA_V) $(VALAC) --compile $(VALA_TESTS_FLAGS) $<
+	$(VALA_V) $(VALAC) --compile --Xcc=-DSECRET_WITH_UNSTABLE=1 $(VALA_TESTS_FLAGS) $<
 
 test_vala_lang_SOURCES =
 test_vala_lang_LDADD = \
-- 
2.9.3

